x-pack/test_serverless/api_integration/test_suites/observability/slos/delete_slo.ts

```diff
@@ -154,19 +152,26 @@ export default function ({ getService }: FtrProviderContext) {
         await transform.api.getTransform(sloTransformId, 404);
         await transform.api.getTransform(sloSummaryTransformId, 404);
 
-        // Roll up and summary data
-        await retry.tryForTime(60 * 1000, async () => {
-          const sloRollupDataAfterDeletion = await sloApi.getSloData({
-            sloId,
-            indexName: SLO_DESTINATION_INDEX_PATTERN,
-          });
+        await retry.waitForWithTimeout('SLO summary data is deleted', 60 * 1000, async () => {
           const sloSummaryDataAfterDeletion = await sloApi.getSloData({
             sloId,
             indexName: SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
           });
+          if (sloSummaryDataAfterDeletion.hits.hits.length > 0) {
+            throw new Error('SLO summary data not deleted yet');
+          }
+          return true;
+        });
 
-          expect(sloRollupDataAfterDeletion.hits.hits.length).to.be(0);
-          expect(sloSummaryDataAfterDeletion.hits.hits.length).to.be(0);
+        await retry.waitForWithTimeout('SLO rollup data is deleted', 60 * 1000, async () => {
+          const sloRollupDataAfterDeletion = await sloApi.getSloData({
+            sloId,
+            indexName: SLO_DESTINATION_INDEX_PATTERN,
+          });
+          if (sloRollupDataAfterDeletion.hits.hits.length > 0) {
+            throw new Error('SLO rollup data not deleted yet');
+          }
+          return true;
```

Comment:
Since we don't wait for completion on the delete of the SLI data, we need to check until it's done or it times out